File: 10.py
def get_primes(m, n):
    bool_arr = [True for i in range(n-1)]
    for i in range(2, n+1):
        for j in [i*i + k for k in range(0, n-i*i+1, i)]:
            bool_arr[j-2] = False
    primes = []
    for i in range(m-2, n-1):
        if bool_arr[i]: primes.append(i+2)
    return primes

m, n = map(int, input().split())
primes = get_primes(m, n)
if len(primes) == 0:
    print('Absent')
else:
    for i in primes: print(i)

File: 10.hs
import Data.Array.Unboxed ( UArray, (!), (//), indices, listArray )

getPrimes :: Int -> Int -> [Int]
getPrimes m n = filter (>= m) primes where
    boolArr :: UArray Int Bool
    boolArr = listArray (2, n) $ replicate (n-1) True
    notPrimes = [j | i <- [2..n], i*i <= n, j <- [i*i+k | k <- [0,i..n-i*i]]]
    boolArr' = boolArr // [(i, False) | i <- notPrimes]
    filterPrimes i is = if boolArr' ! i then i:is else is
    primes = foldr filterPrimes [] $ indices boolArr'

main :: IO ()
main = do
    [m, n] <- map read . words <$> getLine
    let primes = getPrimes m n
    case primes of
        [] -> putStrLn "Absent"
        _ -> mapM_ print primes